scripts/hic/qc_hic.py

```diff
@@ -0,0 +1,153 @@
+        "Reported_pairs",
+        "total_R2",
+        "mapped_R2",
+        "total_R1",
+        "mapped_R1",
+    ]
+    RESULTS = {}
+    for eachfile in FILES:
+        with open(eachfile) as f:
+            for line in f:
+                array = line.split()
+                if array[0] in VARIABLE:
+                    RESULTS[array[0]] = int(array[1])
+
+    PERCENTAGES = {}
+    COUNTS = {}
+    VERDICT = {}
+    REP_VAR = [
+        "R1_aligned",
+        "R2_aligned",
+        "valid_interactionPairs",
+        "cis_shortRange",
+        "cis_longRange",
+    ]
+
+    if args.fithichip_bed is not None and os.path.isfile(args.fithichip_bed):
+        with open(args.fithichip_q01_bed) as fithichip:
+            LOOPS_SIGNIFICANT = len(fithichip.readlines()) - 1
+        with open(args.fithichip_bed) as fithichip:
+            LOOPS = len(fithichip.readlines()) - 1
+    if args.peaks_bed is not None and os.path.isfile(args.peaks_bed):
+        with open(args.peaks_bed) as peaks:
+            PEAKS = len(peaks.readlines()) - 1
+
+    PERCENTAGES["R1_aligned"] = round(
+        (RESULTS["mapped_R1"] * 100) / RESULTS["total_R1"]
+    )
+    PERCENTAGES["R2_aligned"] = round(
+        (RESULTS["mapped_R2"] * 100) / RESULTS["total_R2"]
+    )
+    PERCENTAGES["valid_interactionPairs"] = round(
+        (RESULTS["valid_interaction"] * 100) / RESULTS["Reported_pairs"]
+    )
+    PERCENTAGES["cis_shortRange"] = round(
+        (RESULTS["cis_shortRange"] * 100) / RESULTS["valid_interaction"]
+    )
+    PERCENTAGES["cis_longRange"] = round(
+        (RESULTS["cis_longRange"] * 100) / RESULTS["valid_interaction"]
+    )
+    COUNTS["R1_aligned"] = RESULTS["mapped_R1"]
+    COUNTS["R2_aligned"] = RESULTS["mapped_R2"]
+    COUNTS["valid_interactionPairs"] = RESULTS["valid_interaction"]
+    COUNTS["cis_shortRange"] = RESULTS["cis_shortRange"]
+    COUNTS["cis_longRange"] = RESULTS["cis_longRange"]
+
+    for aligned in ["R1_aligned", "R2_aligned"]:
+        if PERCENTAGES[aligned] > 80:
+            VERDICT[aligned] = "GOOD"
+        else:
+            VERDICT[aligned] = "BAD"
+
+    if PERCENTAGES["valid_interactionPairs"] > 50:
+        VERDICT["valid_interactionPairs"] = "GOOD"
+    else:
+        VERDICT["valid_interactionPairs"] = "BAD"
+
+    if PERCENTAGES["cis_shortRange"] > 50:
+        VERDICT["cis_shortRange"] = "BAD"
+    elif PERCENTAGES["cis_shortRange"] > 30:
+        VERDICT["cis_shortRange"] = "MARGINAL"
+    else:
+        VERDICT["cis_shortRange"] = "GOOD"
+
+    if PERCENTAGES["cis_longRange"] > 40:
+        VERDICT["cis_longRange"] = "GOOD"
+    elif PERCENTAGES["cis_longRange"] > 20:
+        VERDICT["cis_longRange"] = "MARGINAL"
+    else:
+        VERDICT["cis_longRange"] = "BAD"
+
+    REPORT = open(args.prefix + "_QCreport.txt", "w")
+    REPORT.write("STAT\tCOUNTS\tPERCENTAGE\tVERDICT\n")
+    REPORT.write(
+        "Total_pairs_processed\t" + str(RESULTS["Total_pairs_processed"]) + "\n"
+    )
```

Comment:
We probably want to keep the same format as the OG script. But also this is a strange format. I guess just complaining, we shouldn't modify this I don't think

Reply:
Yeah, this tripped me up when writing a test. I assumed there were two empty cells (`\t\t\n`) rather than just a `\n`.